stock_reserve_sale/wizard/sale_stock_reserve.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2013 Camptocamp SA - Guewen Baconnier
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, exceptions, fields, models
+
+
+class SaleStockReserve(models.TransientModel):
+    _name = "sale.stock.reserve"
+    _description = "Sale Stock Reserve"
+
+    @api.model
+    def _default_location_id(self):
+        domain = [
+            "|",
+            ("company_id", "=", self.env.company.id),
+            ("company_id", "=", False),
+        ]
+        return self.env["stock.warehouse"].search(domain, limit=1).lot_stock_id
+
+    @api.model
+    def _default_location_dest_id(self):
+        return self.env["stock.reservation"]._default_location_dest_id()
+
+    def _default_owner(self):
+        """If sale_owner_stock_sourcing is installed, it adds an owner field
+        on sale order lines. Use it.
+
+        """
+        model = self.env[self.env.context["active_model"]]
+        if model._name == "sale.order":
+            lines = model.browse(self.env.context["active_id"]).order_line
+        else:
+            lines = model.browse(self.env.context["active_ids"])
+
+        try:
+            owners = {line.stock_owner_id for line in lines}
+        except AttributeError:
+            return self.env["res.partner"]
+            # module sale_owner_stock_sourcing not installed, fine
+
+        if len(owners) == 1:
+            return owners.pop()
+        elif len(owners) > 1:
+            raise exceptions.Warning(
+                _(
+                    """The lines have different owners. Please reserve them
+                    individually with the reserve button on each one."""
+                )
+            )
+
+        return self.env["res.partner"]
+
+    location_id = fields.Many2one(
```

Comment:
The only con here is that there is no way now to have multiple reservations per one order line as you cannot indicate the quantity. I would add, as an improvement commit, the possibility to indicate the quantity you want to reserve for the line (having all the quantities left to be reserved as default).